src/app.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable camelcase */
+import { Bytes, KeyType, PrivateKey, PublicKey } from '@greymass/eosio';
+import randomBytes from 'randombytes';
+import { ES256KSigner, createJWT } from 'did-jwt';
+import * as jose from 'jose';
+import { PublicKey as PublicKeyCon } from 'eosjs/dist/eosjs-key-conversions';
+function generateRandomKeyPair(): { privateKey: PrivateKey; publicKey: PublicKey } {
+    const bytes = randomBytes(32);
+    const privateKey = new PrivateKey(KeyType.K1, new Bytes(bytes));
+    const publicKey = privateKey.toPublic();
+    return { privateKey, publicKey };
+}
+
+async function onPressLogin(redirect: string): Promise<string> {
```

Comment:
rename `redirect` to 'origin' and give example of origin e.g. 'https://netflix.com`
origin = protocol + domain + port
this is what we need to securely verify for the redirect

Reply:
actually, send in the `window` object and an optional boolean if the user should be redirected
```ts
async function onPressLogin(window: Window, redirect = false): Promise<string> {
...
origin = window.location.origin
...
if (redirect) {
 window.location = settings.loginUrl
}
...
```

something like this